lib/mayaUsd/commands/layerEditorCommand.cpp

```diff
@@ -136,6 +137,18 @@ class InsertRemoveSubPathBase : public BaseCmd
             }
             _subPath = layer->GetSubLayerPaths()[_index];
             holdOnPathIfDirty(layer, _subPath);
+
+            // if the layer to remove is the current edit target,
+            // set the root layer as the current edit target
+            auto subLayerHandle = SdfLayer::FindRelativeToLayer(layer, _subPath);
+            auto stage = getStage();
+            auto currentTarget = stage->GetEditTarget().GetLayer();
+            if (currentTarget
+                && currentTarget->GetIdentifier() == subLayerHandle->GetIdentifier()) {
+                _isEditTarget = true;
+                stage->SetEditTarget(stage->GetRootLayer());
```

Comment:
Would be worth considering matching the logic with base proxy shape, ie fallback to session layer of root layer can’t be edited or we forced session layer with option var.